crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -111,9 +117,10 @@ impl FunctionBuilder {
     pub(crate) fn array_constant(
         &mut self,
         elements: im::Vector<ValueId>,
-        element_types: Rc<CompositeType>,
+        // arrays are homogenous, so there is a single type that all elements abide by
+        element_type: Rc<CompositeType>,
```

Comment:
This is not true. Arrays may have multiple element types whenever we have an array of structs or tuples.

```rs
let foo = [(1i8, [2, 3]), (4, [5, 6]), (7, [8, 9])];
```
Is represented in SSA as:
```rs
[i8 1, [Field 2, Field 3], i8 4, [Field 5, Field 6], i8 7, [Field 8, Field 9]]
```